include/fiction/algorithms/simulation/sidb/quickexact.hpp

```diff
@@ -0,0 +1,477 @@
+                // "real" defects assigned but in order to set some SiDBs with a fixed negative charge, this way of
+                // implementation is chosen.
+                for (const auto& cell : detected_negative_sidbs)
+                {
+                    charge_lyt_new.assign_defect(cell, sidb_defect{sidb_defect_type::UNKNOWN, -1,
+                                                                   charge_lyt_new.get_phys_params().epsilon_r,
+                                                                   charge_lyt_new.get_phys_params().lambda_tf});
+                }
+                for (const auto& [cell, defect] : real_placed_defects)
+                {
+                    charge_lyt_new.assign_defect(cell, defect);
+                }
+                // Update all local potentials, system energy and physically validity. Flag is set to "false" to allow
+                // dependent cell to change its charge state based on the N-1 SiDBs to fulfill the local population
+                // stability at its position.
+
+                // False declares that the dependent cell is updated based on the local potential at the position.
+                charge_lyt_new.update_after_charge_change(false);
+
+                // If no positively charged DB can occur in the layout, this scope is executed.
+                if (!three_state_simulation_required)
+                {
+                    simulation_result.additional_simulation_parameters.emplace_back("base_number",
+                                                                                    static_cast<uint64_t>(2));
+                    this->two_state_simulation(charge_lyt_new, simulation_result);
+                }
+
+                // If positively charged DBs can occur in the layout, 3-state simulation is conducted.
+                else
+                {
+                    simulation_result.additional_simulation_parameters.emplace_back("base_number",
+                                                                                    static_cast<uint64_t>(3));
+                    this->three_state_simulation(charge_lyt_new, simulation_result);
+                }
+            }
+
+            // In the case with only one SiDB in the layout (due to external potentials or defects, this single SiDB can
+            // be neutrally or even positively charged.)
+            else if (number_of_SiDBs == 1)
+            {
+                if (three_state_simulation_required)
+                {
+                    charge_lyt.set_base_number(3);
+                }
+                else
+                {
+                    charge_lyt.set_base_number(2);
+                }
+
+                while (charge_lyt.get_charge_index().first < charge_lyt.get_max_charge_index())
+                {
+
+                    if (charge_lyt.is_physically_valid())
+                    {
+                        charge_distribution_surface<Lyt> charge_lyt_copy{charge_lyt};
```

Comment:
Same for `_copy`

Reply:
But in this case it is not more than an actual copy of the layout?

Reply:
But doesn't it have a purpose? Other than being a copy?